Sources/GRPC/AsyncAwaitSupport/GRPCAsyncRequestStream.swift

```diff
@@ -16,12 +16,19 @@
 
 #if compiler(>=5.6)
 
+import NIOCore
+
 /// This is currently a wrapper around AsyncThrowingStream because we want to be
 /// able to swap out the implementation for something else in the future.
 @available(macOS 10.15, iOS 13, tvOS 13, watchOS 6, *)
 public struct GRPCAsyncRequestStream<Element: Sendable>: AsyncSequence {
   @usableFromInline
-  internal typealias _WrappedStream = PassthroughMessageSequence<Element, Error>
+  internal typealias _WrappedStream = NIOThrowingAsyncSequenceProducer<
```

Comment:
I think we should use the non-throwing variant here. The only thing we finish the source with is the `CancellationError` which I think is wrong to use here. `CancellationError`s should be used if the consuming `Task` gets cancelled; however, in our case we are finishing the source. @glbrntt WDYT?

Reply:
[Discussed offline] We not only throw on source finish, but also expose any other errors (see `makeAndInvoke(call:)` in `GRPCAsyncBidirectionalStreamingCall` and `GRPCAsyncServerStreamingCall`); for this, we'll keep the throwing version of the sequence producer but stop finishing the source with `CancellationError` from the server side.